codalab/rest/bundle.py

```diff
@@ -33,21 +567,34 @@ def get_blob(uuid, path=''):
         filename = target_info['name']
 
     if target_info['type'] == 'directory':
+        if byte_range:
+            abort(httplib.BAD_REQUEST, 'Range not supported for directory blobs.')
+        if head_lines:
+            abort(httplib.BAD_REQUEST, 'Head not supported for directory blobs.')
         # Always tar and gzip directories.
         filename = filename + '.tar.gz'
         fileobj = local.download_manager.stream_tarred_gzipped_directory(uuid, path)
     elif target_info['type'] == 'file':
+        gzipped = False
         if not zip_util.path_is_archive(filename) and request_accepts_gzip_encoding():
             # Let's gzip to save bandwidth. The browser will transparently decode
             # the file.
             filename = filename + '.gz'
-            fileobj = local.download_manager.stream_file(uuid, path, gzipped=True)
+            gzipped = True
+
+        if byte_range and (head_lines or tail_lines):
+            abort(httplib.BAD_REQUEST, 'Head and range not supported on the same request.')
+        elif byte_range:
+            start, end = byte_range
```

Comment:
nit: Range queries with start and length are easier to understand since you don't have to specify whether end is inclusive or exclusive.